### ISSUE TITLE
Formatting guidelines for Cargo.toml?

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang-nursery/fmt-rfcs#5]
This seems incredibly minor, but I wanted to record it for consideration.

Should the fmt process also provide guidelines for formatting of Cargo.toml, or potentially other standard files in the Rust ecosystem other than .rs files?

Such guidelines would likely not need to cover much, but I can think of a few cases that could probably use guidance, especially if we can support it in `rustfmt` and `cargo fmt`.

**Comment by steveklabnik** [comment]
One case that comes up:

```toml
[dependencies]
bar = "1.0.0"
foo = {path = "some/path/to/foo" }
```

vs


```toml
[dependencies.bar]
version = "1.0.0"

[dependencies.foo]
path = "some/path/to/foo"
```

or, my favorite


```toml
[dependencies]
bar = "1.0.0"

[dependencies.foo]
path = "some/path/to/foo"
```

That is, when only depending on a version, put them all under `[dependencies]`, but when doing something more complex, give them their own block. I like this style because it keeps it minimal for minimal cases, but is easier to read for more complex ones.

**Comment by joshtriplett** [comment]
I prefer to still use a braced block for git or path dependencies, because then I can still keep all the dependencies sorted alphabetically.

Speaking of which, two formatting guidelines here:
- Keep all dependencies sorted alphabetically.
- In the `[package]` section, put `name` and `version` first, then sort the remaining keys alphabetically.

**Comment by IvanUkhov** [comment]
It might be also helpful to settle on indentation. For instance,

```toml
[package]
authors = ["foo <foo>",
           "bar <bar>"]
```

vs.

```toml
[package]
authors = [
  "foo <foo>",
  "bar <bar>",
]
```

vs.

```toml
[package]
authors = [
    "foo <foo>",
    "bar <bar>",
]
```

**Comment by joshtriplett** [comment]
I prefer the third of those: the same 4-space indentation as Rust code, and block indent rather than visual indent.

(Though those only apply if the line gets too long.)

**Comment by nrc** [comment]
I'm not entirely what to do here. It does sound like it is probably worth having some guidelines. On the other hand Rustfmt does not support this and it would be a bit tricky to add (and, IMO, low priority). There also seems not enough in this thread to justify having an FCP for.

Therefore, I think that if @joshtriplett (or someone else) wants to write up a PR which adds some short guidelines to the style guide, then that would be an optimal solution. Otherwise, I'll close this in a month or two.

**Comment by joshtriplett** [comment]
@nrc Done, in https://github.com/rust-lang-nursery/fmt-rfcs/pull/41

**Comment by nrc** [comment]
Closed by #41

**Comment by softprops** [comment]
Did support for this every make it into rustfmt proper?
